### PR TITLE
admission: validate scale of deployment

### PR DESCRIPTION
**Pull request by daichimukai**

### CODE REVIEW

File: admission/config/webhook/manifests.yaml [hunk 1]
```diff
@@ -115,6 +115,26 @@ webhooks:
...
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /validate-scale-deployment-replica-count
+  failurePolicy: Fail
+  name: vscaledeploymentreplicacount.kb.io
+  rules:
+  - apiGroups:
+    - apps
```
Comment (peng225) [inline admission/config/webhook/manifests.yaml:130]: ここはscaleではなくdeploymentのGroup/Versionを書くというので正しいのでしょうか？

**Comment by daichimukai** [comment]
- update TAG file
- write document

File: admission/hooks/validate_deployment_replica_count.go [hunk 2]
```diff
@@ -51,3 +52,38 @@ func (v *deploymentReplicaCountValidator) Handle(ctx context.Context, req admiss
...
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+func NewDeploymentReplicaCountScaleValidator(c client.Client, decoder *admission.Decoder) http.Handler {
+	return &webhook.Admission{Handler: &deploymentReplicaCountScaleValidator{client: c, decoder: decoder}}
+}
+
+func (v *deploymentReplicaCountScaleValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	s := &autoscalingv1.Scale{}
```
Comment (daichimukai) [inline admission/hooks/validate_deployment_replica_count.go:68]:
```suggestion
	scale := &autoscalingv1.Scale{}
```

File: admission/hooks/validate_deployment_replica_count.go [hunk 3]
```diff
@@ -51,3 +52,38 @@ func (v *deploymentReplicaCountValidator) Handle(ctx context.Context, req admiss
...
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	deploy := &appsv1.Deployment{}
+	err = v.client.Get(ctx, client.ObjectKey{Namespace: s.GetNamespace(), Name: s.GetName()}, deploy)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if val, ok := deploy.GetAnnotations()[annotationForForceReplicaCount]; !ok || val != "0" {
```
Comment (daichimukai) [inline admission/hooks/validate_deployment_replica_count.go:80]: split to a function

File: admission/config/webhook/manifests.yaml (hunk 1 above)
Comment (pluser) [reply admission/config/webhook/manifests.yaml:130]: Documents: 
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request

**Review by peng225 — APPROVED** [review]

**Review by morimoto-cybozu — APPROVED** [review]
LGTM

**Review by pluser — APPROVED** [review]

**Pull request merged**